src/m365/spo/commands/site/site-recyclebinitem-move.spec.ts

```diff
@@ -0,0 +1,210 @@
+  });
+
+  it('aborts moving the items to the second-stage recycle bin when confirm option not passed and prompt not confirmed', async () => {
+    const postSpy = sinon.spy(request, 'post');
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: false }
+    ));
+    await command.action(logger, {
+      options: {
+        siteUrl: 'https://contoso.sharepoint.com',
+        all: true
+      }
+    });
+
+    assert(postSpy.notCalled);
+  });
+
+  it('moves all items to the second-stage recycle bin with ids and confirm option', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/recycleBin('85528dee-00d5-4c38-a6ba-e2abace32f63')/MoveToSecondStage`) {
+        return '';
+      }
+
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/recycleBin('aecb840f-20e9-4ff8-accf-5df8eaad31a1')/MoveToSecondStage`) {
+        return '';
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.doesNotReject(command.action(logger, {
+      options: {
+        verbose: true,
+        siteUrl: 'https://contoso.sharepoint.com',
+        ids: '85528dee-00d5-4c38-a6ba-e2abace32f63, aecb840f-20e9-4ff8-accf-5df8eaad31a1',
+        confirm: true
+      }
+    }));
+  });
+
+  it('moves all items to the second-stage recycle bin with all option', async () => {
+    const postSpy = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/recycleBin/MoveAllToSecondStage`) {
+        return '';
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    await command.action(logger, {
+      options: {
+        siteUrl: 'https://contoso.sharepoint.com',
+        all: true
+      }
+    });
+
+    assert(postSpy.called);
+  });
+
+  it('moves all items to the second-stage recycle bin with all and confirm option', async () => {
+    const postSpy = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/_api/web/recycleBin/MoveAllToSecondStage`) {
+        return '';
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        verbose: true,
+        siteUrl: 'https://contoso.sharepoint.com',
+        all: true,
+        confirm: true
+      }
+    });
+
+    assert(postSpy.called);
+  });
+
+  it('handles error correctly', async () => {
+    const error = {
+      error: {
+        message: "The files cannot be moved to the second-stage recycle bin"
+      }
+    };
+
+    sinon.stub(request, 'post').callsFake(async () => {
+      throw error;
+    });
+
+    await assert.rejects(command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com', all: true, confirm: true } } as any), error.error);
```

Comment:
We have to wrap the error message in a `CommandError` object. Right now the test is succeeding while the error message doesn't match.

Reply:
I can't wrap it in a `CommandError` object, because it makes the test fail. But I was able to fix it with the actual message

Reply:
We are not fixing the test by just adding the actual message. This means that the test will never fail, which is undesired behavior. There's probably something wrong with the error object you are returning.